internal/observer/mcp/handlers.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2025 The OpenChoreo Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package mcp
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/openchoreo/openchoreo/internal/observer/opensearch"
+	"github.com/openchoreo/openchoreo/internal/observer/service"
+)
+
+type MCPHandler struct {
+	Service *service.LoggingService
+}
+
+// GetComponentLogs retrieves logs for a specific component
+func (h *MCPHandler) GetComponentLogs(params opensearch.ComponentQueryParams) (string, error) {
+	ctx := context.Background()
```

Comment:
shouldn't this ctx get passed down from the request?

Reply:
Good catch. The context should be passed down from the MCP SDK --> Tool handlers --> MCP handler functions. Updated the PR.